Bar gauge card feature for sensors

MindFreeze opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue or discussion
  in the additional information section.
-->
Very basic for now. Only for % sensors.

<img width="990" height="246" alt="image" src="https://github.com/user-attachments/assets/3d9d598a-73f6-4d27-8c98-78531b6652d2" />


## Type of change
<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/orgs/home-assistant/discussions/489 https://github.com/orgs/home-assistant/discussions/67
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/40495

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

karwosts committed:
Force dark-only themes to be in dark mode (#26583)

karwosts committed:
Fetch translations for switch-as-x domains (#26566)

* Fetch translations for switch-as-x domains

* no cache

silamon committed:
Migrate ha-progress-ring to webawesome (#26542)

silamon committed:
Migrate all sl-animation usages to wa-animation (#26544)

wendevlin committed:
Remove unused shoelace animation import (#26584)

MindFreeze committed:
Bar graph card feature for sensors

MindFreeze committed:
use state color

dependabot[bot] committed:
Bump actions/checkout from 4.2.2 to 5.0.0 (#26586)

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.2.2 to 5.0.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4.2.2...v5.0.0)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: 5.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

timmo001 committed:
Add date card feature (#26531)

* Add date card feature

* Rename from "date" to "date-set"

* Better label

* Fix

* Parse date for datetime

karwosts committed:
Show accurate state of show_header_toggle in entities card editor (#26564)

timmo001 committed:
Add clipboard support to condition card conditions (#26535)

* Add copy paste support to condition card conditions

* Don't clear the clipboard

* Fix selection

* Add cut and duplicate

* Remove

* Fix event

MindFreeze committed:
tweak name

karwosts committed:
Fix picture glance mouse pointer (#26391)

* Fix picture glance mouse pointer

* More action updates

MindFreeze committed:
rename and improve colors

NoRi2909 committed:
Capitalize "Samba" as the name of the Open Source project (#26590)

Capitalize "Samba" as name for the Open Source project

renovate[bot] committed:
Update dependency @codemirror/language to v6.11.3 (#26589)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

timmo001 committed:
Set default forecast option in weather forecast card editor (#26592)

pcan08 committed:
Add fan oscillate feature (#26519)

* First working fan-oscillate feature

This a first working impl, need at least to do:
- Tooltip not yet "Yes/No"
- Need implementation verification

* Use same strings as more info label for control tooltip

* Add missing label for editor

* Rename some variables

* Add fan features in gallery

* Fix lint:types by applying suggestions from code review

Co-authored-by: Aidan Timson <aidan@timmo.dev>

* fix lint new line after import

* fix typo

Co-authored-by: Petar Petrov <MindFreeze@users.noreply.github.com>

* fix event value type treating

* remove type magic as suggested

Co-authored-by: Petar Petrov <MindFreeze@users.noreply.github.com>

* Update localize.ts

Complete suggestion change to have tooltip

* fix lint by removing unused import

---------

Co-authored-by: Aidan Timson <aidan@timmo.dev>
Co-authored-by: Petar Petrov <MindFreeze@users.noreply.github.com>

renovate[bot] committed:
Lock file maintenance (#26598)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

karwosts committed:
Fix legend in devices-detail-graph (#26596)

balloob committed:
Change Suggest with AI label to Suggest (#26603)

wendevlin committed:
Automation row remember collapsed status (#26604)

MindFreeze commented:
"Bar gauge" sounds good to me

pcan08 commented:
Hi there,
 
if I may, I also find it not easy to reach the good wording, so, I also ask to IA to give its opinion, which is **LevelBar**.

What do you think about it ?

piitaya commented on src/panels/lovelace/card-features/hui-progress-bar-card-feature.ts:
```diff
@@ -0,0 +1,87 @@
+    }
+    :host > div {
+      height: 100%;
+      background-color: var(--feature-color);
```
```suggestion
      background-color: var(--feature-color);
      transition: width 180ms ease-in-out;
```

piitaya reviewed (commented):
Code wise, it's ok for me. We need to agree on a name and have UX feedback.

piitaya committed:
Summaries overview dashboard (#26594)

* Change categories to lights, climate and security

* Add entities grouped by devices

* Fix garage cover device class

* Rename category to summary

* Add media players categories

* Reduce spacing

* Remove person

* Remove translations

* Use media player cards

* Display entities without device

* Add missing entity category

* Update src/panels/lovelace/strategies/overview/helpers/overview-summaries.ts

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

* Add white space

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

MindFreeze committed:
Update src/panels/lovelace/card-features/hui-progress-bar-card-feature.ts

Co-authored-by: Paul Bottein <paul.bottein@gmail.com>

Drinory committed:
Fix entities not showing due to JavaScript crash (fixes #25363) (#26599)

* Fix entity UI crash from undefined entity names (fixes #25363)

* Fix mock function type compatibility in test

- Update mock to handle string | undefined parameter
- Maintain test functionality while satisfying type checker

* Simplify approach based on reviewer feedback

- Use String() coercion to preserve numeric entity names (e.g., power strip outlets)
- Single line change instead of complex type validation across multiple files
- Revert stripPrefixFromEntityName to original (no longer needs null handling)
- Remove separate test file, update existing test to expect stringified numbers
- More conservative approach that preserves data rather than replacing with fallbacks

MindFreeze committed:
rename to Bar gauge

NoRi2909 reviewed (commented):
Three leftovers from the old name.

marcinbauer85 commented:
While I like the initial simplicity of it, because we already have similar UI components, this is too similar to our current brightness feature on lights:
<img width="544" height="205" alt="CleanShot 2025-08-20 at 10 24 46" src="https://github.com/user-attachments/assets/7b3fb557-f4a8-4957-ac64-2c7a0a3fcecf" />
@MindFreeze I cannot guarantee I'll be able to help with this before beta, because we have other priorities.

MindFreeze committed:
History chart card feature (#26555)

Co-authored-by: Bram Kragten <mail@bramkragten.nl>

balloob committed:
Fix space in overview welcome caption (#26618)

* Fix space in overview welcome caption

* Update src/panels/lovelace/strategies/overview/overview-home-view-strategy.ts

MindFreeze committed:
name fix

MindFreeze committed:
Merge branch 'dev' into feature-bar

MindFreeze commented:
> I cannot guarantee I'll be able to help with this before beta, because we have other priorities.

In that case lets merge this now and we can improve it later.
For the record I think the handle is enough to differentiate this read only feature from the input slider and it is not a big deal if you mistake them anyway. But if a better design comes up, we can implement it later.
